doc/nrf/app_dev/device_guides/working_with_nrf/nrf54h/ug_nrf54h20_suit_sdfw_sdrfw_update.rst

```diff
@@ -0,0 +1,54 @@
+In order to ensure that a trusted execution can be established on the device, a valid Secure Domain Firmware Recovery has to be present on the device prior to performing an Secure Domain Firmware update.
+An update is started only if the update candidate can be verified by Secure Domain ROM using a key matching the requested update type.
+
+
+SUIT attempts to provide flexibility with regard to launching and updating firmware while allowing certain parts of the code base to remain unmodifiable for end users. Although there are various implementations possible, a platform-specific layer that maps logical firmware blocks to their corresponding memory ranges of a device is a straightforward approach for simple systems. The manifest(s) can describe actions needed to launch or update such firmware block.
+For certain areas, nrf54 family uses such approach, however Secure Domain Firmware and Secure Domain Firmware Recovery cannot be simply copied to destination address in order to replace the old firmware - its design relies on Secure Domain ROM launching and updating the binaries.
+
+
+Implementation
+**************
+
+SUIT manifests utilize a concept of components that describe executable/updatable entities. For a given platform there can be a variety of components with different characteristics that require different approach for launching or updating given component (e.g. execution in place versus copying to RAM, etc.).
+
+As described in :ref:`suit_component_types`, there is a special SUIT component type ``SOC_SPEC``
+available for describing components that are controlled by Nordic.
+
+For nRF54H20, there are two components of this type, one for Secure Domain Firmware and the
+other one for Secure Domain Recovery Firmware.
+They are distinguished by component IDs of 1 and 2, respectively:
+
+* ``SOC_SPEC/1`` - Secure Domain Firmware
+
+* ``SOC_SPEC/2`` - Secure Domain Firmware Recovery
+
+As shown in :ref:`suit_default_manifest_topology_for_the_nrf54h20_soc` both components are part of a
+single SUIT manifest. The manifest is controlled by ``Nordic top`` manifest, which in turn is
+controlled by the ``Root`` manifest.
+
+Update process
+**************
+
+Design decision that Secure Domain Firmware and Secure Domain Firmware Recovery are controlled by a single manifest has impact on how these two slots are updated.
+When update of one of the slots fails while the other one succeeds, it could lead to ambiguity of system state.
+Because of that, it was decided that manifest version will reflect Secure Domain Firmware version, as the more crucial slot of the two.
+Secure Domain Firmware Recovery update is attempted first and when it fails, it should not prevent attempt of Secure Domain Firmware update.
```

Comment:
```suggestion
The update process begins by attempting to update the Secure Domain Firmware (SDFW) recovery image.
If this update fails, the process still proceeds with updating the SDFW.
To prevent ambiguity if one slot update fails while the other succeeds, the manifest version reflects the version of the Secure Domain Firmware.
```